schema_utils.py

```diff
@@ -80,7 +80,11 @@ def normalize_against_schema(obj, schema, apply_custom_validators=True):
         from core.domain import obj_services  # pylint: disable=relative-import
         obj_class = obj_services.Registry.get_object_class_by_type(
             schema[SCHEMA_KEY_OBJ_TYPE])
-        normalized_obj = obj_class.normalize(obj)
+        if not apply_custom_validators:
```

Comment:
Why does this general code need to be changed?

Note that apply_custom_validators is handled further down the function, for all values.

Reply:
`obj_class.normalize` again calls `schema_utils.normalize_against_schema` as seen [here](https://github.com/oppia/oppia/blob/develop/extensions/objects/models/objects.py#L57). Here `cls.schema` is `unicode schema` due to [this](https://github.com/oppia/oppia/pull/6286/files#diff-0524862ee50cd155533a74585f4bc036R30), obj is empty string as default value is empty. Eariler, `apply_custom_validator` was `True` in this call. Now due to [if](https://github.com/oppia/oppia/pull/6286/files#diff-781314b975a52120938e342ae5ffd787R83), if `apply_custom_validator` was `False` initially, it is false for this call also.

Reply:
I think I'm confused. SanitizedUrl doesn't have type "custom". It has type "unicode".

So why does this branch of the code even run in this case?

Reply:
`url` is custom [here](https://github.com/oppia/oppia/blob/develop/assets/rich_text_components_definitions.js#L94) with obj_type SanitizedUrl.